// sportmonks/dist/core.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SportMonksCore = void 0;
const client_1 = require("./client");
class SportMonksCore extends client_1.HttpClient {
    constructor(conf) {
        super(`${client_1.API_BASE_URL}`, conf.apiToken, conf.timezone);
        this.conf = conf;
    }
    get(path, opts) {
        return super.get(`/v3/core${path}`, opts);
    }
    continents(opts) {
        return this.get('/continents', opts);
    }
    continentsById(id, opts) {
        return this.get(`/continents/${id}`, opts);
    }
    countries(opts) {
        return this.get('/countries', opts);
    }
    countriesById(id, opts) {
        return this.get(`/countries/${id}`, opts);
    }
    searchCountries(name, opts) {
        return this.get(`/countries/search/${name}`, opts);
    }
    regions(opts) {
        return this.get('/regions', opts);
    }
    regionsById(id, opts) {
        return this.get(`/regions/${id}`, opts);
    }
    searchRegions(name, opts) {
        return this.get(`/regions/search/${name}`, opts);
    }
    cities(opts) {
        return this.get('/cities', opts);
    }
    citiesById(id, opts) {
        return this.get(`/cities/${id}`, opts);
    }
    searchCities(name, opts) {
        return this.get(`/cities/search/${name}`, opts);
    }
    types(opts) {
        return this.get('/types', opts);
    }
    typesById(id, opts) {
        return this.get(`/types/${id}`, opts);
    }
}
exports.SportMonksCore = SportMonksCore;
