// sportmonks/dist/odds.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SportMonksOdds = void 0;
const client_1 = require("./client");
class SportMonksOdds extends client_1.HttpClient {
    constructor(conf) {
        super(`${client_1.API_BASE_URL}`, conf.apiToken, conf.timezone);
        this.conf = conf;
    }
    get(path, opts) {
        return super.get(`/v3/odds${path}`, opts);
    }
    // markets
    markets(opts) {
        return this.get(`/markets`, opts);
    }
    marketsById(id, opts) {
        return this.get(`/markets/${id}`, opts);
    }
    marketsSearchByName(name, opts) {
        return this.get(`/markets/search/${name}`, opts);
    }
    // bookmakers
    bookmakers(opts) {
        return this.get(`/bookmakers`, opts);
    }
    bookmakersById(id, opts) {
        return this.get(`/bookmakers/${id}`, opts);
    }
    bookmakersSearchByName(name, opts) {
        return this.get(`/bookmakers/search/${name}`, opts);
    }
    bookmakersByFixtureId(id, opts) {
        return this.get(`/bookmakers/fixtures/${id}`, opts);
    }
    bookmakersMappingByFixtureId(id, opts) {
        return this.get(`/bookmakers/fixtures/${id}/mapping`, opts);
    }
}
exports.SportMonksOdds = SportMonksOdds;
